pulp_rpm/tests/functional/api/test_crud_content_unit.py

```diff
@@ -126,15 +133,31 @@ def test_04_delete(self):
         self.assertIn(msg, exc.exception.args[0])
 
     @skip_if(bool, "content_unit", False)
-    def test_05_duplicate_raise_error(self):
-        """Attempt to create duplicate package."""
+    def test_05_duplicate_no_repo_return_existing_href(self):
+        """Attempt to create duplicate package without specifying a repository."""
         attrs = gen_rpm_content_attrs(self.artifact, RPM_PACKAGE_FILENAME)
         response = self.rpm_content_api.create(**attrs)
-        with self.assertRaises(PulpTaskError) as cm:
-            monitor_task(response.task)
-        task_result = cm.exception.task.to_dict()
-        msg = "There is already a package with"
-        self.assertTrue(msg in task_result["error"]["description"])
+        duplicate = self.rpm_content_api.read(monitor_task(response.task).created_resources[0])
+        assert duplicate.pulp_href == self.content_unit["pulp_href"]
+
+    @skip_if(bool, "content_unit", False)
+    def test_06_duplicate_add_to_repo(self):
+        """Attempt to create duplicate package while specifying a repository."""
+        repo = self.repo_api.create(gen_repo())
+        attrs = gen_rpm_content_attrs(self.artifact, RPM_PACKAGE_FILENAME)
+        attrs["repository"] = repo.pulp_href
+        response = self.rpm_content_api.create(**attrs)
+        monitored_response = monitor_task(response.task)
+
+        duplicate = self.rpm_content_api.read(monitored_response.created_resources[1])
```

Comment:
In general, this all looks correct.  However this is a bit of a philosophical question - should we be returning the content unit as a "created resource" when we did not create it?  I can see arguments either way.

On one hand, on its face it's not correct to say that we're creating this resource.  But on the other hand, it likely doesn't matter very much to the user, so we might just create more work for the API consumer (diifferent paths to handle whether it did or didn't get returned) for not a lot of practical benefit.

Also this is similar to two other issues, both of which might be addressed by this effort:  

https://github.com/pulp/pulpcore/issues/2911
https://github.com/pulp/pulpcore/issues/1914

Reply:
@mdellweg What do you think?

Reply:
Considering RBAC, the user might not even be able to *see* the content before uploading themselves. So It would even reveal (sensitive) information if we didn't act as they just created it.

(Instead of "created_resources", I would rather prefer if tasks were able to just return something.)

Reply:
Well that raises another problem, if you aren't the first person to upload a package, do you just never get to access it?

Reply:
>Considering RBAC, the user might not even be able to see the content before uploading themselves. So It would even reveal (sensitive) information if we didn't act as they just created it.

Ah, my 8 AM brain didn't quite get what you were saying here, but I understand now.

Reply:
In any case @MichalPysik I recommend taking a look at the two other issues and determining if they can be closed as duplicates or wontfix on the basis of the changes you are making.